File: pilhas/Pilha.py
class Pilhas:
    def __init__(self) -> None:
        self.__pilha = list()
        
    def adiciona(self, item):
        self.__pilha.append(item)
        
    def pop(self):
        if self.isEmpty():
            raise Exception("A pilha está vazia.")
        ultimoIndex = self.__getUltimoElemento()
        return self.__pilha.pop(ultimoIndex)
        
    
    def peek(self):
        if self.isEmpty():
            raise Exception("A pilha está vazia.")
        ultimoIndex = self.__getUltimoElemento()
        return self.__pilha[ultimoIndex]
    
    def isEmpty(self):
        if(self.__pilha == []):
            return True
        return False;
    
    def size(self):
        return len(self.__pilha)
    
    def clear(self):
        self.__pilha = list()
    
    def __getUltimoElemento(self):
        return len(self.__pilha) - 1
    
    def __str__(self) -> str:
        return str(self.__pilha)

File: conteudos/pilhas/Pilha.java
package pilhas;

import java.util.ArrayList;

public class Pilha<E> {
    private ArrayList<E> pilha;

    public Pilha() {
        this.pilha = new ArrayList<>();
    }

    public void adicionar(E item) {
        this.pilha.add(item);
    }

    public E pop(){
        if (this.isEmpty()) {
            throw new IllegalStateException("A pilha está vazia.");
        }
        int ultimoIndex = getUltimoIndex();
        return this.pilha.remove(ultimoIndex);
    }

    public E peek(){
        if (this.isEmpty()) {
            throw new IllegalStateException("A pilha está vazia.");
        }
        int ultimoIndex = getUltimoIndex();
        return this.pilha.get(ultimoIndex);
    }

    public boolean isEmpty(){
        if (this.pilha.isEmpty()) {
            return true;
        }
        return false;
    }

    public int size(){
        return this.pilha.size();
    }

    public void clear(){
        this.pilha = new ArrayList<>();
    }

    private int getUltimoIndex(){
        return this.pilha.size() - 1;
    }

    @Override
    public String toString(){
        return this.pilha.toString();
    }
}
